processing/src/main/java/io/druid/segment/data/GenericIndexed.java

```diff
@@ -413,11 +421,11 @@ public int size()
 
     T bufferedIndexedGet(ByteBuffer copyValueBuffer, int startOffset, int endOffset)
     {
-      final int size = endOffset - startOffset;
+      final int size = endOffset > startOffset
```

Comment:
It passes -1 as numBytes down into ObjectStrategies. I suggest to not do so.
```java
int size = endOffset - startOffset;
if (size == 0 && copyValueBuffer.get(startOffset - Ints.BYTES) == NULL_VALUE_SIZE_MARKER) {
  return null;
}
```
etc.

Reply:
done.